src/eventcontroller/DefaultEventController.ts

```diff
@@ -108,28 +175,47 @@ export default class DefaultEventController implements EventController {
       name: state,
       timestampMs,
     });
+    // Reports event to the ingestion service
     this.reportEvent(state, timestampMs);
   }
 
   private getAttributes(timestampMs: number): EventAttributes {
     return {
-      attendeeId: this.audioVideoController.configuration.credentials.attendeeId,
+      attendeeId: this.configuration.credentials.attendeeId,
       browserMajorVersion: this.browserMajorVersion,
       browserName: this.browserName,
       browserVersion: this.browserVersion,
       deviceName: this.deviceName,
       externalMeetingId:
-        typeof this.audioVideoController.configuration.externalMeetingId === 'string'
-          ? this.audioVideoController.configuration.externalMeetingId
+        typeof this.configuration.externalMeetingId === 'string'
+          ? this.configuration.externalMeetingId
           : '',
-      externalUserId: this.audioVideoController.configuration.credentials.externalUserId,
+      externalUserId: this.configuration.credentials.externalUserId,
       meetingHistory: this.meetingHistoryStates,
-      meetingId: this.audioVideoController.configuration.meetingId,
+      meetingId: this.configuration.meetingId,
       osName: this.parserResult?.os.name || DefaultEventController.UNAVAILABLE,
       osVersion: this.parserResult?.os.version || DefaultEventController.UNAVAILABLE,
       sdkVersion: Versioning.sdkVersion,
       sdkName: Versioning.sdkName,
       timestampMs,
     };
   }
+
+  get eventReporter(): EventReporter {
+    return this._eventReporter;
+  }
+
+  /**
+   * Clean up this instance and resources that it created.
+   *
+   * After calling `destroy`, internal fields like `eventReporter` will be unavailable.
+   */
+  async destroy(): Promise<void> {
+    if (isDestroyable(this.eventReporter)) {
+      await this.eventReporter.destroy();
```

Comment:
Based on earlier comments, update `eventReporter` usage to `this._eventReporter`. Or use `private eventReporter?: EventReporter` in constructor and then just use `eventReporter`.

Reply:
As mentioned before this works because there is a getter.